### PR TITLE
Remove 46 projects that were migrated to the rushstack-legacy repo

### PR DESCRIPTION
**Pull request by octogonz**
## Summary

The `gulp-core-build-*` and `rushstack-compiler-*` projects are no longer actively developed.
They have been moved to the [rushstack-legacy](https://github.com/microsoft/rushstack-legacy) GitHub repo.

### CODE REVIEW

File: apps/rush-lib/src/cli/test/basicAndRunBuildActionRepo/b/package.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "name": "b",
   "version": "1.0.0",
   "description": "Test package b",
+  "dependencies": {
```
Comment (iclanton) [inline apps/rush-lib/src/cli/test/basicAndRunBuildActionRepo/b/package.json:5]: Unrelated?

**Review by iclanton — APPROVED** [review]

File: apps/rush-lib/src/cli/test/basicAndRunBuildActionRepo/b/package.json (hunk 1 above)
Comment (octogonz) [reply apps/rush-lib/src/cli/test/basicAndRunBuildActionRepo/b/package.json:5]: It's a fix for a test failure that turned out to be a race condition.  Making `b` dependent on `a` forces the build to happen in a specific order.
